Cards/CardContainer.gd

```diff
@@ -74,10 +79,23 @@ func set_active_card(card: CardBase) -> void:
 	_active_card = card
 
 
-func remove_active_card(card: CardBase) -> void:
+func finish_active_card_action(card: CardBase) -> void:
+	_discard_active_card()
 	_active_card = null
 
 
+func _discard_active_card() -> void:
+	var _card_to_discard = null
+	for card_index in _cards_queued_for_discard.size():
```

Comment:
I would change this for loop into a custom find function so you can return the result instead of continuing to loop.